ddl/db_legacy_rename_test.go

```diff
@@ -0,0 +1,374 @@
+// Copyright 2022 PingCAP, Inc.
```

Comment:
I don't think this test should be named with `legacy` but `db_rename_test.go`.

Reply:
Also in https://github.com/pingcap/tidb/issues/28636 I split tasks by suite dimension so that you can migrate one suite by the other, which its member tests should be dispatched into different tests. In this way we are naturally work on disjoint subset and never conflict with each other.

This PR is good to go, but later you may think of this division.

Reply:
We can be late to merge this PR and split the test into the different files that their filename are called to ```db_suite_{}_test.go```.

Reply:
NO. I don't want to create any file named as `db_suite_n_test.go`, it's meaning less. See also #32955 for example, I just dispatch test into other file or create a new one with meaningful name if necessary.